etc/config/command.php

```diff
@@ -4,34 +4,84 @@
  * See COPYING.txt for license details.
  */
 
-if (isset($_POST['command'])) {
+if (isset($_POST['baseUrl']) && isset($_POST['username']) && isset($_POST['password']) && isset($_POST['command'])) {
+    $baseUrl = urldecode($_POST['baseUrl']);
+    $username = urldecode($_POST['username']);
+    $password = urldecode($_POST['password']);
     $command = urldecode($_POST['command']);
     if (array_key_exists("arguments", $_POST)) {
         $arguments = urldecode($_POST['arguments']);
     } else {
         $arguments = null;
     }
-    $php = PHP_BINDIR ? PHP_BINDIR . '/php' : 'php';
-    $valid = validateCommand($command);
-    if ($valid) {
-        exec(
-            escapeCommand($php . ' -f ../../../../bin/magento ' . $command) . " $arguments" ." 2>&1",
-            $output,
-            $exitCode
-        );
-        if ($exitCode == 0) {
-            http_response_code(202);
+
+    if (isAuthenticated($baseUrl, $username, $password)) {
+        $php = PHP_BINDIR ? PHP_BINDIR . '/php' : 'php';
+        $magentoBinary = $php . ' -f ../../../../bin/magento';
+        $valid = validateCommand($magentoBinary, $command);
+        if ($valid) {
+            exec(
+                escapeCommand($magentoBinary . ' ' . $command) . " $arguments" ." 2>&1",
+                $output,
+                $exitCode
+            );
+            if ($exitCode == 0) {
+                http_response_code(202);
+            } else {
+                http_response_code(500);
+            }
+            echo implode("\n", $output);
         } else {
-            http_response_code(500);
+            http_response_code(403);
+            echo "Given command not found valid in Magento CLI Command list.";
         }
-        echo implode("\n", $output);
     } else {
-        http_response_code(403);
-        echo "Given command not found valid in Magento CLI Command list.";
+        http_response_code(401);
+        echo("Command not unauthorized.");
     }
 } else {
     http_response_code(412);
-    echo("Command parameter is not set.");
+    echo("Required parameters are not set.");
+}
+
+/**
+ * Returns if credentials are successfully authenticated.
+ *
+ * @param string $baseUrl
+ * @param string $username
+ * @param string $password
+ * @return bool
+ */
+function isAuthenticated($baseUrl, $username, $password)
```

Comment:
I would like to try avoiding performing a CURL request within this file if possible, and instead would rather leverage the fact that this file exists side-by-side with Magento itself to authenticate whatever we pass in.

I left a comment on the ticket with some sample code.